### PR TITLE
[googlemaps] geocoder JSDoc & base test

### PR DESCRIPTION
**Pull request by demensky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/maps/documentation/javascript/reference/geocoder
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/googlemaps/test/reference/geocode-tests.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Geocoder = google.maps.Geocoder;
+
+const geocoder = new Geocoder();
+
+geocoder.geocode({}, (results, status) => {
+    results; // $ExpectType GeocoderResult[]
+    status; // $ExpectType GeocoderStatus
+});
+
+export {};
```
Comment (peterblazejewicz) [inline types/googlemaps/test/reference/geocode-tests.ts:10]:
```suggestion
```
IMO this module exports nothing, just consumes

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@demensky thx!
wait and see for owners

File: types/googlemaps/test/reference/geocode-tests.ts (hunk 1 above)
Comment (demensky) [reply types/googlemaps/test/reference/geocode-tests.ts:10]: There is no `import` or `export` in the file ([namespace imports](https://www.typescriptlang.org/docs/handbook/namespaces.html#aliases) are not counted). If I don't add `export {}`, then TypeScript will store all variables declared here and treat them as global. The rest of the test files use a similar technique.

Comment (peterblazejewicz) [reply types/googlemaps/test/reference/geocode-tests.ts:10]: but this is a test file, for compiler related checks, it doesn't get published (as type file that is) nor being consumed at all at any stage

Comment (peterblazejewicz) [reply types/googlemaps/test/reference/geocode-tests.ts:10]: oh, I see, that's becuase of global exports in the DT details, so in tests aliases are used. Sry!

Comment (demensky) [reply types/googlemaps/test/reference/geocode-tests.ts:10]: example1
example2

**Review by amcasey — APPROVED** [review]

**Comment by demensky** [comment]
Ready to merge

**Pull request merged**